src/main/java/org/checkstyle/autofix/CheckstyleRecipeRegistry.java

```diff
@@ -47,28 +55,55 @@ private CheckstyleRecipeRegistry() {
      * using the simple name of the check, and applies the factory to generate Recipe instances.
      *
      * @param violations the list of Checkstyle violations
+     * @param config the checkstyle configuration
      * @return a list of generated Recipe objects
      */
-    public static List<Recipe> getRecipes(List<CheckstyleViolation> violations) {
-
-        final Map<String, List<CheckstyleViolation>> violationsByCheck = violations.stream()
-                .collect(Collectors.groupingBy(CheckstyleViolation::getSource));
-
-        final List<Recipe> recipes = new ArrayList<>();
+    public static List<Recipe> getRecipes(List<CheckstyleViolation> violations,
+                                          CheckConfiguration config) {
+        return violations.stream()
+                .collect(Collectors.groupingBy(CheckstyleViolation::getSource))
+                .entrySet()
+                .stream()
+                .map(entry -> createRecipe(entry, config))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
 
-        for (Map.Entry<String, List<CheckstyleViolation>> entry : violationsByCheck.entrySet()) {
-            final String checkName = entry.getKey();
-            final String simpleCheckName = checkName
-                    .substring(checkName.lastIndexOf('.') + 1);
-            final List<CheckstyleViolation> checkViolations = entry.getValue();
+    private static Recipe createRecipe(Map.Entry<String, List<CheckstyleViolation>> entry,
+                                       CheckConfiguration config) {
+        final String simpleCheckName = normalizeCheckName(entry.getKey());
+        final List<CheckstyleViolation> violations = entry.getValue();
 
-            final Function<List<CheckstyleViolation>, Recipe> recipeFactory =
+        Recipe recipe = null;
+        final BiFunction<List<CheckstyleViolation>, CheckConfiguration, Recipe> configFactory =
```

Comment:
I know that some people prefer explicit `if`, but I am more in favor of chaining :) 

ex. 

what do you think about this modification?

```
        return Optional.ofNullable(RECIPE_MAP_WITH_CONFIG.get(simpleCheckName))
                .map(factory -> factory.apply(violations, extractCheckConfiguration(config, simpleCheckName)))
                .orElseGet(() -> Optional.ofNullable(RECIPE_MAP.get(simpleCheckName))
                        .map(factory -> factory.apply(violations))
                        .orElse(null));
```